Cache/Requests.lean

```diff
@@ -28,10 +28,10 @@ section Get
 
 /-- Formats the config file for `curl`, containing the list of files to be downloaded -/
 def mkGetConfigContent (hashMap : IO.HashMap) : IO String := do
-  let l ← hashMap.foldM (init := []) fun acc _ hash => do
+  hashMap.foldM (init := "") fun acc _ hash => do
     let fileName := hash.asTarGz
-    pure $ (s!"url = {← mkFileURL fileName none}\n-o {IO.CACHEDIR / fileName}") :: acc
-  return "\n".intercalate l
+    pure $ acc ++ s!"url = {← mkFileURL fileName none}\n-o {
+      (IO.CACHEDIR / fileName).toString.quote}\n"
```

Comment:
```suggestion
    -- Below we use `String.quote`, which is intended for quoting for use in Lean code
    -- this does not exactly match the requirements for quoting for curl:
    -- ```
    -- If the parameter contains whitespace (or starts with : or =),
    --  the parameter must be enclosed within quotes. 
    -- Within double quotes, the following escape sequences are available:
    --  \, ", \t, \n, \r and \v. 
    -- A backslash preceding any other letter is ignored.
    -- ```
    -- If this becomes an issue we can implement the curl spec.
    pure $ acc ++ s!"url = {← mkFileURL fileName none}\n-o {
      (IO.CACHEDIR / fileName).toString.quote}\n"
```
This is a maximally verbose comment noting Eric's concern. I'm happy with a much shorter one, too. :-)